state/payloads.go

```diff
@@ -160,6 +178,25 @@ func (up UnitPayloads) Untrack(name string) error {
 	return nil
 }
 
+var checkPayloadClassForUnit = validatePayloadClassForUnit
+
+func validatePayloadClassForUnit(unit *Unit, c, t string) error {
```

Comment:
this needs to return []txn.Op to assert that the charm has not changed, using Unit.assertCharmOps

rather than do this validation here, you should probably update payload{Track,Untrack,SetStatus}Change.Prepare  to call it, and include the ops in their results